lib/EarlyFunctionAnalysis/EarlyFunctionAnalysis.cpp

```diff
@@ -878,6 +914,54 @@ static MetaAddress getFinalAddressOfBasicBlock(llvm::BasicBlock *BB) {
   return End + Size;
 }
 
+UpcastablePointer<efa::FunctionEdgeBase>
+FunctionEntrypointAnalyzer::handleFunctionCallForCFG(llvm::CallInst *Call) {
+  using namespace llvm;
+
+  MetaAddress Destination;
+  Value *SymbolNameString = nullptr;
+  efa::FunctionEdgeType::Values Type;
+  auto *CalleePC = Call->getArgOperand(1);
+
+  // Direct or indirect call?
+  if (isa<ConstantStruct>(CalleePC)) {
+    auto AddressPC = MetaAddress::fromConstant(CalleePC);
+    // Does the function exist within the model?
+    auto It = Binary->Functions.find(AddressPC);
+    if (It != Binary->Functions.end()) {
+      Destination = AddressPC;
+      Type = efa::FunctionEdgeType::FunctionCall;
+    } else {
+      // Does the function exist within the imported functions?
+      auto *OriginalEntry = GCBI->getBlockAt(AddressPC);
+      SymbolNameString = hasDynamicSymbol(OriginalEntry);
+      if (SymbolNameString) {
+        // Dynamic function call. We set `Edge.Destination` to the callee's
+        // MetaAddress, and fix it to `Invalid` later in `finalizeModel`.
+        Destination = AddressPC;
+        Type = efa::FunctionEdgeType::FunctionCall;
```

Comment:
This seems plain wrong. You should not set `CallEdge::Destination` but `CallEdge::DynamicFunction`.
This is quite fundamental.

Reply:
Fixed this by anticipating `CE->DynamicFunction = SymbolName` from `finalizeModel` to `collectDirectCFG`. `finalizeModel`  accomplishes the creation of the prototype for indirect calls (that ought to be done there).